datahub-web-react/src/app/search/filters/utils.tsx

```diff
@@ -0,0 +1,297 @@
+    if (activeFilters.find((activeFilter) => activeFilter.field === filterField)) {
+        newFilters = activeFilters
+            .map((f) => (f.field === filterField ? { ...f, values: selectedFilterValues } : f))
+            .filter((f) => !(f.values?.length === 0));
+    } else {
+        newFilters = [...activeFilters, { field: filterField, values: selectedFilterValues }].filter(
+            (f) => !(f.values?.length === 0),
+        );
+    }
+    return newFilters;
+}
+
+export function isFilterOptionSelected(selectedFilterOptions: FilterOptionType[], filterValue: string) {
+    const parentFilterValues = filterValue.includes(FILTER_DELIMITER)
+        ? filterValue.split(FILTER_DELIMITER).slice(0, -1)
+        : [];
+    return !!selectedFilterOptions.find(
+        (option) => option.value === filterValue || parentFilterValues.includes(option.value),
+    );
+}
+
+export function isAnyOptionSelected(selectedFilterOptions: FilterOptionType[], filterValues?: string[]) {
+    return selectedFilterOptions.some((option) => filterValues?.some((filterValue) => filterValue === option.value));
+}
+
+export function getFilterEntity(filterField: string, filterValue: string, availableFilters: FacetMetadata[] | null) {
+    return (
+        availableFilters
+            ?.find((facet) => facet.field === filterField)
+            ?.aggregations.find((agg) => agg.value === filterValue)?.entity || null
+    );
+}
+
+export const PlatformIcon = styled.img<{ size?: number }>`
+    max-height: ${(props) => (props.size ? props.size : 12)}px;
+    width: auto;
+    object-fit: contain;
+    background-color: transparent;
+`;
+
+function getDataPlatformInstanceIconAndLabel(
+    filterEntity: Entity | null,
+    entityRegistry: EntityRegistry,
+    size?: number,
+) {
+    let icon: React.ReactNode = null;
+    let label: React.ReactNode = null;
+    const logoUrl = (filterEntity as DataPlatformInstance)?.platform.properties?.logoUrl;
+    icon = logoUrl ? (
+        <PlatformIcon src={logoUrl} size={size} />
+    ) : (
+        entityRegistry.getIcon(EntityType.DataPlatform, size || 12, IconStyleType.ACCENT, ANTD_GRAY[9])
+    );
+    label = (filterEntity as DataPlatformInstance).instanceId
+        ? (filterEntity as DataPlatformInstance).instanceId
+        : (filterEntity as DataPlatformInstance).urn;
+
+    return { icon, label };
+}
+
+export function getLastBrowseEntryFromFilterValue(filterValue: string) {
+    const browseEntries = filterValue.split(UNIT_SEPARATOR);
+    return browseEntries[browseEntries.length - 1] || '';
+}
+
+export function getFilterIconAndLabel(
+    filterField: string,
+    filterValue: string,
+    entityRegistry: EntityRegistry,
+    filterEntity: Entity | null,
+    size?: number,
+    filterLabelOverride?: string | null,
+) {
+    let icon: React.ReactNode = null;
+    let label: React.ReactNode = null;
+
+    if (filterField === ENTITY_FILTER_NAME || filterField === LEGACY_ENTITY_FILTER_NAME) {
```

Comment:
nit: there is a lot of nested conditional logic. in general, i'd recommend trying to split out some of the insides of these conditional blocks into separate functions